GoogleDataTransport/GDTCORLibrary/GDTCOREvent.m

```diff
@@ -75,15 +75,17 @@ - (void)setDataObject:(id<GDTCOREventDataObject>)dataObject {
   }
 }
 
-- (void)setFileURL:(NSURL *)fileURL {
-  if (fileURL != _fileURL) {
-    _fileURL = fileURL;
-  }
-}
-
-- (void)clearDataObjectTransportBytes {
-  if (_dataObjectTransportBytes != nil) {
-    _dataObjectTransportBytes = nil;
+- (void)writeToURL:(NSURL *)fileURL error:(NSError **)error {
+  if (_dataObjectTransportBytes) {
+    BOOL writingSuccess = [_dataObjectTransportBytes writeToURL:fileURL
+                                                        options:NSDataWritingAtomic
+                                                          error:error];
+    if (!writingSuccess) {
+      GDTCORLogError(GDTCORMCEFileWriteError, @"An event file could not be written: %@", fileURL);
+    } else {
+      _fileURL = fileURL;
```

Comment:
If writing fails, we should probably assign nil to _fileURL since that file doesn't exist